src/test/java/com/company/services/TwitterServiceTest.java

```diff
@@ -0,0 +1,121 @@
+
+public class TwitterServiceTest {
+
+    private Twitter twitterMock;
+    private TwitterAppConfigurationKeys mockKeys;
+    private TwitterService twitterService;
+
+    @Before
+    public void setup() {
+        twitterMock = mock(Twitter.class);
+        mockKeys = new TwitterAppConfigurationKeys();
+        twitterService = TwitterService.getInstance();
+        twitterService.setTwitter(twitterMock);
+    }
+
+    @Test
+    public void tweetGood() throws TwitterException {
+        when(twitterMock.updateStatus(anyString())).thenReturn(getFixtureStatus("I am a fixture status"));
+        TwitterPost expected = getExpectedTwitterPost("I am a fixture status");
+        assertEquals(expected, twitterService.postTweet("this is a good string!", mockKeys));
+    }
+
+    @Test(expected = TwitterException.class)
+    public void tweetTooLong() throws TwitterException {
+        String val = RandomString.make(281);
+        when(twitterMock.updateStatus(val)).thenThrow(new TwitterException("must be less than 280 chars"));
+        twitterService.postTweet(val, mockKeys);
+    }
+
+    @Test(expected = TwitterException.class)
+    public void tweetIsBlank() throws TwitterException {
+        String val = "";
+        when(twitterMock.updateStatus(val)).thenThrow(new TwitterException("cannot be null"));
+        twitterService.postTweet(val, mockKeys);
+    }
+
+    @Test(expected = TwitterException.class)
+    public void tweetIsNull() throws TwitterException {
+        String val = null;
+        when(twitterMock.updateStatus(val)).thenThrow(new TwitterException("cannot be null"));
+        twitterService.postTweet(val, mockKeys);
+    }
+
+    @Test
+    public void timelineGood() throws TwitterException {
+        ResponseList<Status> fixtureResponseList = new FixtureResponseList<>();
+        fixtureResponseList.add(getFixtureStatus("I am a fixture status"));
+        when(twitterMock.getHomeTimeline()).thenReturn(fixtureResponseList);
+        List<TwitterPost> expected = new ArrayList<>();
+        expected.add(getExpectedTwitterPost("I am a fixture status"));
+        assertEquals(expected, twitterService.getTimeline(mockKeys));
+    }
+
+    @Test
+    public void filteredTimelineGood() throws TwitterException {
+        ResponseList<Status> fixtureResponseList = new FixtureResponseList<>();
+        fixtureResponseList.add(getFixtureStatus("Hello is in this one"));
+        fixtureResponseList.add(getFixtureStatus("But not in this one"));
+        when(twitterMock.getHomeTimeline()).thenReturn(fixtureResponseList);
+        List<TwitterPost> expected = new ArrayList<>();
+        expected.add(getExpectedTwitterPost("Hello is in this one"));
+        assertEquals(expected, twitterService.getFilteredTimeline("hello", mockKeys));
+        expected.add(getExpectedTwitterPost("But not in this one"));
+        assertEquals(expected, twitterService.getFilteredTimeline("one", mockKeys));
+    }
+
+    @Test(expected = TwitterException.class)
```

Comment:
i would generally test the exception message as well in case there are different ones passed back